### PR TITLE
research page tweaks on staging

### PR DESCRIPTION
**Pull request by ethn**

### CODE REVIEW

File: mod/research/set/self/research_page/left_research_side.haml [hunk 1]
```diff
@@ -14,6 +14,13 @@
         = button_tag "View Methodology", class: "_methodology-tab btn btn-sm btn-outline-secondary"
     .large-text
       = nest [metric, :question], view: :core
-- if answer?
-  .p-2.pl-4
+.p-2.pl-4
+  - if answer?
     = answer_slot
+  - else
```
Comment (ethn) [inline mod/research/set/self/research_page/left_research_side.haml:20]: Prompt to select field if answer form not ready.

File: mod/research/set/self/research_page/_year_option.haml [hunk 2]
```diff
@@ -1,11 +1,12 @@
...
-      %strong
-        = item
-    -if record_card.answers_by_year[item]
-      %div
-        = nest record_card.answers_by_year[item], view: :conciser, hide: :year_equals
-    -else
-      %i
-        add answer
\ No newline at end of file
+- if item
```
Comment (ethn) [inline mod/research/set/self/research_page/_year_option.haml:1]: In dropdown, first year wasn't selected and so wasn't triggering change.  This let me add a blank year.

File: mod/research/lib/stylesheets/style_research.scss [hunk 3]
```diff
@@ -113,20 +113,25 @@ form.answer-form {
...
     top: 117px;
-}
+  }
+  .details, .details.stick {
+    overflow: hidden;
+    width: 58.3333333333%;
+    left: 41.6666666667%;
+    position: fixed;
+    max-width: none;
+  }
```
Comment (ethn) [inline mod/research/lib/stylesheets/style_research.scss:127]: fixed "jumping" sources.

File: mod/metrics/set/abstract/metric_answer/source_views/source_selector.haml [hunk 4]
```diff
@@ -1,3 +1,4 @@
-= render :sourcebox
-= render :my_sources
-= render :suggested_sources
+.source-selector{ class: ("no-citation" if card.year == "false") }
```
Comment (ethn) [inline mod/metrics/set/abstract/metric_answer/source_views/source_selector.haml:1]: don't show cite button when answer from isn't ready.

**Review by xithan — APPROVED** [review]

File: lib/capistrano/tasks/deploy.cap [hunk 5]
```diff
@@ -119,38 +119,39 @@ namespace :deploy do
...
-      else
-        puts "Could not find #{story.id} in git repo."
-      end
-    end
-  end
+  # after :cleanup, :deploy_pt_stories do
+  #   pt_config_file = File.expand_path("../../../../config/pivotaltracker.yml", __FILE__)
+  #   next if fetch(:stage) != :staging || !File.exists?(pt_config_file)
+  #
+  #   require 'pivotal-tracker'
```
Comment (xithan) [inline lib/capistrano/tasks/deploy.cap:126]: you could replace `pivotal-tracker` with `tracker-api`
https://github.com/dashofcode/tracker_api

File: lib/capistrano/tasks/deploy.cap [hunk 6]
```diff
@@ -119,38 +119,39 @@ namespace :deploy do
...
+  #     raise "#{n} not set in pivotaltracker.yml" unless tracker[n]
+  #   end
+  #
+  #   PivotalTracker::Client.token = tracker[:token]
+  #   PivotalTracker::Client.use_ssl = true
+  #
+  #   require "pry"
+  #   unpakt_project = PivotalTracker::Project.find(tracker[:project_id])
+  #   stories =
+  #     unpakt_project.stories.all(:state => "finished", :story_type => ['bug', 'feature'])
```
Comment (xithan) [inline lib/capistrano/tasks/deploy.cap:138]: probably `project.stories(filter: "state:finished type:bug,feature")` will do the same
PT filter api: https://www.pivotaltracker.com/help/articles/advanced_search/

File: lib/capistrano/tasks/deploy.cap [hunk 7]
```diff
@@ -119,38 +119,39 @@ namespace :deploy do
...
+  #
+  #   staging_deploy_tag = `git tag | grep staging | tail -n1`
+  #
+  #   stories.each do |story|
+  #     next if story.labels&.include? "not merged"
+  #     puts "Searching for #{story.id} in local git repo."
+  #     search_result = `git log --grep #{story.id} #{staging_deploy_tag}`
+  #     if search_result.length > 0
+  #       puts "Found #{story.id}, marking as delivered."
+  #       story.notes.create(:text => "Delivered by staging deploy script.")
```
Comment (xithan) [inline lib/capistrano/tasks/deploy.cap:148]: `story.create_comment(text: "Delivered by ...")`

**Pull request merged**